waspls/src/Wasp/LSP/Core.hs

```diff
@@ -0,0 +1,46 @@
+module Wasp.LSP.Core
+  ( ServerM,
+    Severity (..),
+    State,
+    ServerConfig,
+  )
+where
+
+import Control.Monad.Trans.Except (ExceptT)
+import Control.Monad.Trans.State.Strict (StateT)
+import Data.Aeson
+import Data.Aeson.Types (prependFailure, typeMismatch)
+import Data.Default (Default (def))
+import Data.Text (Text)
+import Language.LSP.Server (LspT)
+
+type ServerM =
+  ExceptT (Severity, Text) (StateT State (LspT ServerConfig IO))
```

Comment:
So right now we can return only one error message? Is that ok?

Reply:
These are separate from diagnostics, and are for errors that happen when the language server is processing a request. One error message should be enough.

Reply:
Could we maybe comment here a bit, to make it clear what this error is then?
We could maybe even introduce smth like `type LSError = (Severity, Text)`, maybe even new `newtype`, if that would help document it.

Reply:
Added `ServerError` type